Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -49,48 +49,47 @@ try {
     }
 
     if ($settings.versioningstrategy -ne -1) {
-        if ($getSettings -contains 'appBuild' -or $getSettings -contains 'appRevision') {
-            switch ($settings.versioningStrategy -band 15) {
-                0 { # Use RUN_NUMBER and RUN_ATTEMPT
-                    $settings.appBuild = $settings.runNumberOffset + [Int32]($ENV:GITHUB_RUN_NUMBER)
-                    $settings.appRevision = [Int32]($ENV:GITHUB_RUN_ATTEMPT) - 1
-                }
-                1 { # Use RUN_ID and RUN_ATTEMPT
-                    OutputError -message "Versioning strategy 1 is no longer supported"
-                }
-                2 { # USE DATETIME
-                    $settings.appBuild = [Int32]([DateTime]::UtcNow.ToString('yyyyMMdd'))
-                    $settings.appRevision = [Int32]([DateTime]::UtcNow.ToString('HHmmss'))
-                }
-                15 { # Use maxValue
-                    $settings.appBuild = [Int32]::MaxValue
-                    $settings.appRevision = 0
-                }
-                default {
-                    OutputError -message "Unknown version strategy $versionStrategy"
-                    exit
-                }
+        switch ($settings.versioningStrategy -band 15) {
+            0 { # Use RUN_NUMBER and RUN_ATTEMPT
+                $settings.appBuild = $settings.runNumberOffset + [Int32]($ENV:GITHUB_RUN_NUMBER)
+                $settings.appRevision = [Int32]($ENV:GITHUB_RUN_ATTEMPT) - 1
+            }
+            1 { # Use RUN_ID and RUN_ATTEMPT
+                OutputError -message "Versioning strategy 1 is no longer supported"
+            }
+            2 { # USE DATETIME
+                $settings.appBuild = [Int32]([DateTime]::UtcNow.ToString('yyyyMMdd'))
+                $settings.appRevision = [Int32]([DateTime]::UtcNow.ToString('HHmmss'))
+            }
+            15 { # Use maxValue
+                $settings.appBuild = [Int32]::MaxValue
+                $settings.appRevision = 0
+            }
+            default {
+                OutputError -message "Unknown version strategy $versionStrategy"
+                exit
             }
         }
     }
 
     $outSettings = @{}
-    $getSettings | ForEach-Object {
-        $setting = $_.Trim()
+    $settings.Keys | ForEach-Object {
```

Comment:
With the new interpretation, settings is always a full settings construct and very few env variables are added.
the full settings is used by all actions and we don't want to describe in the yaml file which settings the actions in this repo would need.
If we don't change this, then CI/CD workflows will have to get all env variables in settings and as variables.

Reply:
That's not what I meant.

Old design:
- if `$get` is not set, then `$env:Settings` is JSON with all the settings.
- if `$get` is set to `"setting1, setting2"`, then `$env:setting1` and `$env:setting1` are set and `$env:Settings` is a JSON with only `setting1` and `setting2` as keys.

New design is:
- if `$get` is not set, then `$env:Settings` is JSON with all the settings.
- if `$get` is set to `"setting1, setting2"`, then `$env:setting1` and `$env:setting1` are set and **`$env:Settings` is JSON with all the settings**.

Is this correct or I got it wrong?
With the new design, `$env:Settings` is always the full JSON, which isn't necessary. And CI/CD probably works as it doesn't use the `get` parameter.

My main concerns here is: unnecessary settings in `$env:Settings` which leads to log pollution (env variables are always printed in the logs when an action is called, whether or not the action uses them). Generally, I prefer having settings in the logs, but only the relevant ones 😄

Reply:
You have it right - env:Settings is always a full settings object now and the reason is that this settings object will be used by all actions.
It is much simpler to have Settings always be the same - and with all environment variables gone - it looks much nicer in the log:
Check out: https://github.com/BusinessCentralApps/buildorder/actions/runs/5722336769/job/15505342235

Also, Imagine that you need a env setting in CI/CD and add a get: - then suddenly everything falls apart because it changes Settings.
Or if you create another Action, which uses Settings then there is a difference how it works when calling it from CreateApp or from CI/CD - because ENV:Settings are different.

All in all - having Settings be settings and just get the values you need for a workflow is a cleaner solution I think,

Reply:
Looks like this now:
https://github.com/BusinessCentralApps/buildorder/actions/runs/5727505155/job/15520103820

![image](https://github.com/microsoft/AL-Go/assets/10775043/76d4521d-df0e-4115-971f-2cd1d387cfe2)

![image](https://github.com/microsoft/AL-Go/assets/10775043/d56f1862-d866-48e3-9087-5bdfedda4867)

Reply:
Alright then. Let's go with that new design: _ReadSettings_ always loads all the settings.

Can we at least mention (maybe in the doc for the `$get` parameter) that `$get` specifies settings to be directly added as env variables.

Reply:
I will make a pass through all README.md for all actions and make some cleanup and list:

- Name and Description
- ENV input (which ENV variables are assumed set when calling the action and description)
- Parameters (description + required or defaulted)
- ENV output (which ENV variables are set by this action)
- OUTPUT (which output variables are set by this action)
-

Reply:
Done